src/OpenLoco/src/GameCommands/Road/RemoveRoad.cpp

```diff
@@ -0,0 +1,262 @@
+#include "Objects/BridgeObject.h"
+#include "Objects/ObjectManager.h"
+#include "Objects/RoadExtraObject.h"
+#include "Objects/RoadObject.h"
+#include "Random.h"
+#include "RemoveRoadStation.h"
+#include "SceneManager.h"
+#include "World/TownManager.h"
+
+namespace OpenLoco::GameCommands
+{
+    using namespace World::TileManager;
+
+    // Copy of `playTrackRemovalSound`
+    static void playRoadRemovalSound(const World::Pos3 pos)
+    {
+        const auto frequency = gPrng2().randNext(17955, 26146);
+        Audio::playSound(Audio::SoundId::demolish, pos, 0, frequency);
+    }
+
+    static World::RoadElement* getRoadElement(const World::Tile& tile, const RoadRemovalArgs& args, uint8_t sequenceIndex, uint8_t flags)
+    {
+        const auto baseZ = args.pos.z / World::kSmallZStep;
+        const auto companyId = SceneManager::isEditorMode() ? CompanyId::neutral : getUpdatingCompanyId();
+
+        for (auto& element : tile)
+        {
+            auto* elRoad = element.as<World::RoadElement>();
+            if (elRoad == nullptr)
+            {
+                continue;
+            }
+            if (elRoad->baseZ() != baseZ)
+            {
+                continue;
+            }
+            if (elRoad->rotation() != args.rotation)
+            {
+                continue;
+            }
+            if (elRoad->sequenceIndex() != sequenceIndex)
+            {
+                continue;
+            }
+            if (elRoad->roadObjectId() != args.objectId)
+            {
+                continue;
+            }
+            if (elRoad->roadId() != args.roadId)
+            {
+                continue;
+            }
+            if (elRoad->isGhost() != ((flags & Flags::ghost) != 0))
+            {
+                continue;
+            }
+            if (elRoad->isAiAllocated() != ((flags & Flags::aiAllocated) != 0))
+            {
+                continue;
+            }
+            // Ghost only as this is checked elsewhere for non-ghost so that
+            // neutral company is always allowed
+            if (((flags & Flags::ghost) != 0) && elRoad->owner() != companyId)
+            {
+                return nullptr;
+            }
+
+            return elRoad;
+        }
+
+        return nullptr;
+    }
+
+    // 0x00477A10
+    static currency32_t roadRemoveCost(const RoadRemovalArgs& args, const World::TrackData::PreviewTrack roadPiece0, const World::Pos3 roadStart, const uint8_t flags)
+    {
+        const auto roadLoc = roadStart + World::Pos3{ Math::Vector::rotate(World::Pos2{ roadPiece0.x, roadPiece0.y }, args.rotation), roadPiece0.z };
+
+        auto tile = World::TileManager::get(roadLoc.x, roadLoc.y);
+        auto* roadElPiece = getRoadElement(tile, args, roadPiece0.index, flags);
+        if (roadElPiece == nullptr)
+        {
+            return 0;
+        }
+
+        currency32_t totalRemovalCost = 0;
+
+        const auto* roadObj = ObjectManager::get<RoadObject>(roadElPiece->roadObjectId());
+        {
```

Comment:
You should skip the cost calc if the owner of the element is neutral (stop you from profitting from neutral roads)